test/shopify-cli/js_system_test.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+require 'project_types/node/test_helper'
+
+module ShopifyCli
+  class JsSystemTest < MiniTest::Test
+    def setup
+      project_context('app_types', 'node')
+      @system = JsSystem.new(ctx: @context)
+
+      @execution_mock = Minitest::Mock.new
+      @execution_mock.expect(:executed, nil)
+    end
+
+    def test_call_executes_yarn_lambda_if_yarn_is_available
+      @system.stubs(:yarn?).returns(true)
+
+      @system.call(yarn: -> { @execution_mock.executed }, npm: -> { raise Exception })
+
+      @execution_mock.verify
+    end
+
+    def test_call_executes_npm_lambda_if_npm_is_available
```

Comment:
misleading description  - we don't ever check that npm is available. We just check that yarn is not available. 
(here and below)

Reply:
I have updated the test names.